crates/local-deployment/src/container.rs

```diff
@@ -355,180 +360,197 @@ impl LocalContainerService {
         let container = self.clone();
         let analytics = self.analytics.clone();
 
-        tokio::spawn(async move {
-            loop {
-                let status_opt = {
-                    let child_lock = {
-                        let map = child_store.read().await;
-                        map.get(&exec_id)
-                            .cloned()
-                            .unwrap_or_else(|| panic!("Child handle missing for {exec_id}"))
-                    };
+        let mut process_exit_rx = self.spawn_os_exit_watcher(exec_id);
 
-                    let mut child_handler = child_lock.write().await;
-                    match child_handler.try_wait() {
-                        Ok(Some(status)) => Some(Ok(status)),
-                        Ok(None) => None,
-                        Err(e) => Some(Err(e)),
+        tokio::spawn(async move {
+            let mut exit_signal_future = exit_signal
+                .map(|rx| rx.map(|_| ()).boxed()) // wait for signal
+                .unwrap_or_else(|| std::future::pending::<()>().boxed()); // no signal, stall forever
+
+            let status_result: std::io::Result<std::process::ExitStatus>;
+
+            // Wait for process to exit, or exit signal from executor
```

Comment:
Can you add a comment explaining why we have this separate wait condition? E.g. explaining that this is to handle the fact that ACP processes do not exit and have to be manually shutdown